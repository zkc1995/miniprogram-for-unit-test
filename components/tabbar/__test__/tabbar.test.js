const simulate = require('miniprogram-simulate');
const path = require('path')

test('test for tabbar click', async () => {
  const id = simulate.load(path.join(__dirname, '../../tabbar/index'))
  const tabbar = simulate.render(id)
  const parent = document.createElement('parent-wrapper')
  tabbar.attach(parent);
  // const tabbar = tabbar.querySelector('.tabbar')
  const tabbarItems = tabbar.querySelectorAll('.tabbar-item')
  const tabbarContent = tabbar.querySelector('.tabbar-content')

  tabbarItems[1].dispatchEvent('tap')
  await simulate.sleep(0)
  expect(tabbar.data.currentKey).toBe(1)
  expect(`当前点击第${tabbar.data.currentKey}个tab项，对应内容为${tabbarContent.dom.innerHTML}`).toMatchSnapshot()

  tabbarItems[2].dispatchEvent('tap')
  await simulate.sleep(0)
  expect(tabbar.data.currentKey).toBe(2)
  expect(`当前点击第${tabbar.data.currentKey}个tab项，对应内容为${tabbarContent.dom.innerHTML}`).toMatchSnapshot()

  tabbarItems[0].dispatchEvent('tap')
  await simulate.sleep(0)
  expect(tabbar.data.currentKey).toBe(0)
  expect(`当前点击第${tabbar.data.currentKey}个tab项，对应内容为${tabbarContent.dom.innerHTML}`).toMatchSnapshot()

  tabbar.detach()
})