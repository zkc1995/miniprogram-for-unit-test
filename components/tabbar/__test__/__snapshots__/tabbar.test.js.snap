// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for tabbar click 1`] = `
"当前点击第1个tab项，对应内容为
    second
  "
`;

exports[`test for tabbar click 2`] = `
"当前点击第2个tab项，对应内容为
    third
  "
`;

exports[`test for tabbar click 3`] = `
"当前点击第0个tab项，对应内容为
    first
  "
`;
